linux/hci/conn.go

```diff
@@ -184,6 +201,81 @@ func (c *Conn) Write(sdu []byte) (int, error) {
 	return sent, nil
 }
 
+func (c *Conn) initPairingContext() {
+	smp := c.smp
+
+	la := c.LocalAddr().Bytes()
+	//assume la is public address for now
+	//todo: figure out a better way
```

Comment:
couldn't we detect public vs random by checking the MSB of the local addr?

Reply:
yeah, forgot about that, good point